sensor/tests/resource/helper.go

```diff
@@ -0,0 +1,289 @@
+package resource
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"net"
+	"os"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/pkg/errors"
+	"github.com/stackrox/rox/generated/internalapi/central"
+	"github.com/stackrox/rox/generated/storage"
+	"github.com/stackrox/rox/pkg/utils"
+	centralDebug "github.com/stackrox/rox/sensor/debugger/central"
+	"github.com/stackrox/rox/sensor/debugger/message"
+	"github.com/stackrox/rox/sensor/kubernetes/client"
+	"github.com/stackrox/rox/sensor/kubernetes/sensor"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials/insecure"
+	"google.golang.org/grpc/test/bufconn"
+	appsV1 "k8s.io/api/apps/v1"
+	v1 "k8s.io/api/core/v1"
+	v12 "k8s.io/api/rbac/v1"
+	_ "k8s.io/client-go/plugin/pkg/client/auth/gcp"
+	"sigs.k8s.io/e2e-framework/klient/conf"
+	"sigs.k8s.io/e2e-framework/klient/decoder"
+	"sigs.k8s.io/e2e-framework/klient/k8s"
+	"sigs.k8s.io/e2e-framework/klient/k8s/resources"
+	"sigs.k8s.io/e2e-framework/klient/wait"
+	"sigs.k8s.io/e2e-framework/klient/wait/conditions"
+	"sigs.k8s.io/e2e-framework/pkg/envconf"
+)
+
+type YamlTestFile struct {
+	Kind string
+	File string
+}
+
+var (
+	NginxDeployment  = YamlTestFile{"Deployment", "nginx.yaml"}
+	NginxRole        = YamlTestFile{"Role", "nginx-role.yaml"}
+	NginxRoleBinding = YamlTestFile{"Binding", "nginx-binding.yaml"}
+)
+
+type TestCallback func(t *testing.T, testContext *TestContext)
+
+type TestContext struct {
+	t               *testing.T
+	r               *resources.Resources
+	env             *envconf.Config
+	fakeCentral     *centralDebug.FakeService
+	centralReceived chan *central.MsgFromSensor
+}
+
+func NewContext(t *testing.T) (*TestContext, error) {
+	envConfig := envconf.New().WithKubeconfigFile(conf.ResolveKubeConfigFile())
+	r, err := resources.New(envConfig.Client().RESTConfig())
+	if err != nil {
+		return nil, err
+	}
+	fakeCentral, startFn := startSensorAndFakeCentral(envConfig)
+	ch := make(chan *central.MsgFromSensor, 100)
+	fakeCentral.OnMessage(func(msg *central.MsgFromSensor) {
+		ch <- msg
+	})
```

Comment:
The `ch` channel is not being closed.

Reply:
I don't think it's a good idea. It's not harmful since the channel is bound to the lifecycle of the program (which is killed at the end of the test anyway). 

I tried simply closing at the end of the test, but then we get a data race because `fakeCentral` is using it. If we close fake central before closing the channel, then the sensor will panic because the gRPC connection went down. And then we're at the same problem as above, which is that sensor doesn't behave as expected when calling `Stop()`.

Reply:
Gotcha. That makes sense.